syntax = "proto3";


package MetaDataManager;

// The greeting service definition.
service MetaDataManagerService {
  // Sends a greeting
  rpc fileAccessRequestHandler (FileAccessRequest) returns (FileAccessResponse) {}
}

/* The request message containing information for file access
 we added re_ipaddr for server to send responce back on another channel.
 One client can send mutiple request hence using request id
*/

message FileAccessRequest {
  uint32 StartByte = 1;
  uint32 EndByte = 2;
  uint64 RequestId = 3;  
  string FileName = 4;
  string ReqIpaddrPort = 5;  
}

/* The response message containing request status and token if success 
*/
message  FileAccessResponse {
  uint64 RequestId = 1;
  string ReqStatus = 2;
  string Token = 3;
}

